### PR TITLE
Kim/propertyImage

### PR DESCRIPTION
**Pull request by dkimnguyen**
property image stuff done

### CODE REVIEW

File: app/scripts/controllers/myRooms.js [hunk 1]
```diff
@@ -25,6 +25,11 @@ angular.module('adminAppFrontendApp').controller('MyRoomsController', function (
         $rootScope.goToPath('/myRooms/' + roomID);
     };
 
+    $scope.changePhoto = function(){
+        //TODO: store new image to s3, delete old image, store new url in firebase
```
Comment (shakeelmohamed) [inline app/scripts/controllers/myRooms.js]: We can comment this out

File: app/scripts/controllers/signup.js [hunk 2]
```diff
@@ -132,6 +159,9 @@ angular.module('adminAppFrontendApp')
                         newProperty.Latitude = geolocation.lat;
                         newProperty.Longitude = geolocation.lon;
                         newProperty.Country = "US";
+                        if (!newProperty.PhotoUrl){
```
Comment (shakeelmohamed) [inline app/scripts/controllers/signup.js]: Add a comment here, saying we're using the exbedia logo as the default URL

File: app/scripts/controllers/signup.js [hunk 3]
```diff
@@ -50,8 +50,35 @@ angular.module('adminAppFrontendApp')
...
 
             addAccountToFirebase(newUser, newProperty);
+            uploadImageTos3();
         };
 
+        function uploadImageTos3(){
+            var ref = new Firebase ("https://test-admin-accounts.firebaseio.com/cred");
+            ref.once("value", function(snapshot){
+                AWS.config = snapshot.val();
+                var bucketName = 'private-property-images';
```
Comment (shakeelmohamed) [inline app/scripts/controllers/signup.js]: Can we pull this outside?

File: app/scripts/controllers/signup.js [hunk 4]
```diff
@@ -50,8 +50,35 @@ angular.module('adminAppFrontendApp')
...
+                var fileChooser = document.getElementById('file');
+                var results = document.getElementById('results');
+                var file = fileChooser.files[0];
+                if(file) {
+                    var params = {Key: file.name, ContentType: file.type, Body: file};
+                    bucket.upload(params, function (err, data) {
+                        if(err) {
+                            console.log(err);
+                        } else {
+                            $scope.newProperty.PhotoUrl = 'https://s3-us-west-2.amazonaws.com/private-property-images/' + file.name;
```
Comment (shakeelmohamed) [inline app/scripts/controllers/signup.js]: use bucketName to build the URL

File: app/scripts/controllers/signup.js [hunk 5]
```diff
@@ -132,6 +159,9 @@ angular.module('adminAppFrontendApp')
                         newProperty.Latitude = geolocation.lat;
                         newProperty.Longitude = geolocation.lon;
                         newProperty.Country = "US";
+                        if (!newProperty.PhotoUrl){
+                            newProperty.PhotoUrl = "https://s3-us-west-2.amazonaws.com/private-property-images/exbedia_icon_square.png";
```
Comment (shakeelmohamed) [inline app/scripts/controllers/signup.js]: use bucketName to build the URL

File: app/views/myRooms.html [hunk 6]
```diff
@@ -4,8 +4,10 @@
   <table class="table">
     <tr>
       <td>
-        <!-- TODO: update with the hotel's image -->
-        <img src="http://media.expedia.com/hotels/1000000/30000/25300/25290/25290_372_b.jpg" class="img-rounded" alt="{{ property.Name }}" height="236">
+        <!--<div class="btn-group-vertical">-->
+          <img ng-src="{{property.PhotoUrl}}" alt"{{property.Name}}" class="exbedia-property-image">
```
Comment (shakeelmohamed) [inline app/views/myRooms.html]: nit: pull this outside the div & unindent

File: app/views/myRooms.html [hunk 7]
```diff
@@ -4,8 +4,10 @@
...
+          <!--<button class="btn btn-success" ng-click="changePhoto()">Change</buttn>-->
```
Comment (shakeelmohamed) [inline app/views/myRooms.html]: Add a todo saying implement change photo functionality

File: app/views/signup.html [hunk 8]
```diff
@@ -1,3 +1,5 @@
+<div id="status"></div>
+<ul id="objects"></ul>
```
Comment (shakeelmohamed) [inline app/views/signup.html]: These look like debug statements, clean these up & make sure they're not referenced in signup.js

File: app/views/signup.html [hunk 9]
```diff
@@ -88,6 +90,8 @@ <h3 class="text-center">Property Information</h3>
             
             <label for="zipcode">Zip Code:</label>
             <input type="text" class="form-control" id="zipcode" name="zicode" ng-model="newProperty.PostalCode" minlength="5", maxlength="5" required>
+            <label for="file">Property photo:</label>
+            <input class="form-control" type="file" id="file" accept="image/*"/>
```
Comment (shakeelmohamed) [inline app/views/signup.html]: nit: add `name="file"`

File: app/index.html [hunk 10]
```diff
@@ -87,6 +87,7 @@
     <script src="bower_components/angularfire/dist/angularfire.js"></script>
     <script src="bower_components/geofire/dist/geofire.js"></script>
     <script src="bower_components/angularGeoFire/src/angularGeoFire.js"></script>
+    <script src="bower_components/aws-sdk-js/dist/aws-sdk.js"></script>
```
Comment (shakeelmohamed) [inline app/index.html]: I think grunt will break this. You'll probably have to pull it out like I did w/ `rsvp.min.js`.
You can test this yourself:

0. run `grunt build`
0. run git diff, you'll probably see this file change
0. run `grunt serve`
0. check the JS console in Chrome

File: app/scripts/controllers/signup.js [hunk 11]
```diff
@@ -50,8 +50,35 @@ angular.module('adminAppFrontendApp')
...
+                var results = document.getElementById('results');
+                var file = fileChooser.files[0];
+                if(file) {
+                    var params = {Key: file.name, ContentType: file.type, Body: file};
+                    bucket.upload(params, function (err, data) {
+                        if(err) {
+                            console.log(err);
+                        } else {
+                            $scope.newProperty.PhotoUrl = 'https://s3-us-west-2.amazonaws.com/private-property-images/' + file.name;
+                            console.log('The URL is', $scope.newProperty.PhotoUrl);
```
Comment (shakeelmohamed) [inline app/scripts/controllers/signup.js]: nit: this log statement isn't super helpful

File: app/scripts/controllers/signup.js [hunk 12]
```diff
@@ -50,8 +50,35 @@ angular.module('adminAppFrontendApp')
...
+                        }
+                    });
+                }
+            }, function(errorObject){
+                console.log("the read failed: " + errorObject.code);
```
Comment (shakeelmohamed) [inline app/scripts/controllers/signup.js]: nit: this log statement isn't super helpful

File: app/index.html (hunk 10 above)
Comment (dkimnguyen) [inline app/index.html]: When I remove the aws-sdk from the code and run <code>grunt build</code> it automatically generate it.

Comment (shakeelmohamed) [inline app/index.html]: ok as long as it works

File: app/scripts/controllers/signup.js (hunk 2 above)
Comment (shakeelmohamed) [inline app/scripts/controllers/signup.js]: @dkimnguyen do this and I'll merge it in

File: app/index.html (hunk 10 above)
Comment (shakeelmohamed) [inline app/index.html]: @dkimnguyen FYI this does not work, running `grunt build` will delete this entry I'll fix it

**Pull request merged**